### PR TITLE
Add Sensor Code

### PR DESCRIPTION
**Pull request by perigoso**
Not sure we want this how it is right now (do we want to have blobs on the repo?), but works for testing the sensors

### CODE REVIEW

File: src/platform/stm32f1/hal/ticks.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+#pragma once
+
+#include "hal/ticks.h"
+#include "util/types.h"
+
+struct ticks_hal_t ticks_hal_init();
+u64 ticks_hal_ticks_ms();
+void ticks_hal_delay_ms(u32 ticks);
+void ticks_hal_delay_us(u32 ticks);
```
Comment (FFY00) [inline src/platform/stm32f1/hal/ticks.h:14]: Do we need to export these functions?

File: src/driver/pixart/pixart_pmw.c [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+struct pixart_pmw_driver_t pixart_pmw_init(u8 *firmware, struct spi_hal_t spi_hal, struct ticks_hal_t ticks_hal)
+{
+	struct pixart_pmw_driver_t driver = {};
+
+	driver.spi_hal = spi_hal;
+	driver.ticks_hal = ticks_hal;
+
+	if (!firmware)
+		/*return NULL;*/ return driver;
```
Comment (FFY00) [inline src/driver/pixart/pixart_pmw.c:73]: Did you forget to remove the comment?

File: src/driver/pixart/pixart_pmw.c [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	driver.spi_hal.select(driver.spi_hal, 0);
+	driver.ticks_hal.delay_ms(1);
+
+	/* Software Reset */
+	pixart_pmw_write(driver, PIXART_PMW_REG_PWR_UP_RST, PIXART_PMW_RESET_CMD);
+	driver.ticks_hal.delay_ms(50);
+
+	/* Check Known ID */
+	u8 pid = pixart_pmw_read(driver, PIXART_PMW_REG_PID);
+	if (!((pid == PIXART_PMW_PID_PMW3360) || (pid == PIXART_PMW_PID_PMW3330) || (pid == PIXART_PMW_PID_PMW3389)))
```
Comment (FFY00) [inline src/driver/pixart/pixart_pmw.c:84]: nitpick: I generally prefer the expanded De Morgan's law version (`pid != PIXART_PMW_PID_PMW3330 && pid != PIXART_PMW_PID_PMW3360 ...`)

**Review by FFY00 — CHANGES_REQUESTED** [review]
Thanks! I have some comments.

I think we should put the proprietary blobs in a separate repo.

**Comment by perigoso** [comment]
I agree, that was something i was considering, do we include them as a submodule? In /external?

File: src/platform/stm32f1/hal/ticks.h (hunk 1 above)
Comment (perigoso) [reply src/platform/stm32f1/hal/ticks.h:14]: The driver needs this, yes, creating a hal is maybe not the ideal method to provide this functionallity

File: src/driver/pixart/pixart_pmw.c (hunk 2 above)
Comment (perigoso) [reply src/driver/pixart/pixart_pmw.c:73]: I did, this was just placeholder, i wanted to somehow deal with failed initializations

File: src/driver/pixart/pixart_pmw.c (hunk 3 above)
Comment (perigoso) [reply src/driver/pixart/pixart_pmw.c:84]: Works for me

**Comment by perigoso** [comment]
depends on #36

**Review by FFY00 — APPROVED** [review]

**Pull request merged**